src/pyasm/command/file_upload.py

```diff
@@ -175,12 +180,38 @@ def _dump_file_to_temp(my):
             return
 
         System().makedirs(dirname)
+   
+        
+        # Determine if base_decode is necessary
+        base_decode = my.base_decode
+        if my.write_mode == "ab":
+            # Check for base_decode indicator file
+            if (os.path.exists("%s.action" % tmp_file_path)):
+                base_decode = True
+        elif base_decode:
+            # Create indicator file if base_decode is necessary
+            decode_action_path = "%s.action" % tmp_file_path
```

Comment:
define this variable decode_action_path a few lines above before the if statement so it can be used in os.path.exists() above